src/components/DefineCompany/OfficeAddresses.vue

```diff
@@ -0,0 +1,518 @@
+<template>
+  <div id="office-addresses">
+    <div v-if="!isEditing">
+      <v-layout row>
+        <v-flex md4><label><strong>Registered Office</strong></label></v-flex>
+        <v-flex md4>
+          <label><strong>Mailing Address</strong></label>
+          <mailing-address
+            :address="mailingAddress"
+            :editing="isEditing"
+            v-if="!isEmptyAddress(mailingAddress)"/>
+          <div v-else>Not entered</div>
+        </v-flex>
+        <v-flex md4>
+          <label><strong>Delivery Address</strong></label>
+          <delivery-address
+            v-if="!isEmptyAddress(deliveryAddress) && !inheritMailingAddress"
+            :address="deliveryAddress"
+            :editing="isEditing"/>
+          <div v-else>
+            <span v-if="isEmptyAddress(deliveryAddress)">Not entered</span>
+            <span v-else>Same as Mailing Address</span>
+          </div>
+        </v-flex>
+      </v-layout>
+      <v-layout row v-if="entityFilter(EntityTypes.BCOMP)" class="mt-4">
+        <v-flex md4><label><strong>Records Office</strong></label></v-flex>
+        <v-flex md4>
+          <label><strong>Mailing Address</strong></label>
+           <mailing-address
+            :address="recMailingAddress"
+            :editing="isEditing"
+            v-if="!inheritRegisteredAddress && !isEmptyAddress(recMailingAddress)"/>
+           <div v-else>
+             <span v-if="isEmptyAddress(recMailingAddress)">Not entered</span>
+             <span v-else>Same as Registered Office</span>
+           </div>
+        </v-flex>
+        <v-flex md4>
+          <label><strong>Delivery Address</strong></label>
+          <delivery-address
+            v-if="!inheritRecMailingAddress && !inheritRegisteredAddress && !isEmptyAddress(recDeliveryAddress)"
+            :address="recDeliveryAddress"
+            :editing="isEditing"/>
+          <div v-else>
+            <span v-if="isEmptyAddress(recDeliveryAddress)">Not entered</span>
+            <span v-else-if="inheritRegisteredAddress">Same as Registered Office</span>
+            <span v-else>Same as Mailing Address</span>
+          </div>
+        </v-flex>
+      </v-layout>
+    </div>
+    <v-card flat v-else>
+      <ul class="list address-list address-form">
+        <!-- Registered Office Section -->
+        <div class="address-edit-header">
+          <label class="address-edit-title">Registered Office</label>
+        </div>
+
+        <!-- Registered Mailing Address -->
+        <li class="address-list-container">
+          <div class="meta-container">
+            <label>Mailing Address</label>
+            <div class="meta-container__inner">
+              <div class="address-wrapper">
+                <mailing-address
+                  :address="mailingAddress"
+                  :editing="isEditing"
```

Comment:
Is editing prop always true? Consider using 'true' instead of variable to be explicit that this value doesn't change here.
Knowing that this value doesn't change here makes it easy to understand this code (imho).

And ditto for other base address components below.